### PR TITLE
AWS: new module ec2_transit_gateway fixes #49376

### LINKED ISSUE #49376: Support for AWS new Transit gateway
##### SUMMARY
AWS recently launched a new type of gateway: transit gateway. This avoid having create 1000 of peering connections between many VPCs. It would be nice to be able to create / configure them from ansible.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME

aws_vpc_transit_gateway (new)
aws_vpc_transit_gateway_facts (new)

### PR DESCRIPTION
**Pull request by BobBoldin**
##### SUMMARY
AWS: new module ec2_transit_gateway Fixes #49376

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME
ec2_transit_gateway

### CODE REVIEW

**Comment by BobBoldin** [comment]
The integration tests will fail until following permissions are added:

        {
            "Sid": "AllowTransitGatewayManagement",
            "Effect": "Allow",
            "Action": [
                "ec2:CreateTransitGateway",
                "ec2:DeleteTransitGateway",
                "ec2:DescribeTransitGateways"
            ],
            "Resource": "*"
        }

**Comment by BobBoldin** [comment]
ready_for_review

File: test/integration/targets/ec2_transit_gateway/tasks/main.yml
Comment (willthames) [inline test/integration/targets/ec2_transit_gateway/tasks/main.yml]: Please uncomment this :)

**Comment by willthames** [comment]
I'll let @mattclay weigh in on this, but I'd be happy to merge this if it used the unsupported alias for now - the test suite passes fine for me locally

**Review by willthames — CHANGES_REQUESTED** [review]
Needs no_log fix but no other issues once that's addressed.

**Comment by BobBoldin** [comment]
ready_for_review

**Review by willthames — APPROVED** [review]

**Comment by willthames** [comment]
Merged, thanks @BobBoldin, that was a flawless attempt for a first module!

**Comment by mattclay** [comment]
The tests for this module have been enabled in CI: https://github.com/ansible/ansible/pull/54619

However, they are currently marked unstable as they sometimes try to delete the transit gateway while it is in the pending state:

https://app.shippable.com/github/ansible/ansible/runs/116823/104/console

**Pull request merged**
